### PR TITLE
Add manually written tests for RegExp `v` flag proposal

### PR DESCRIPTION
**Pull request by mathiasbynens**
This functionality is part of the RegExp `v` flag proposal: https://github.com/tc39/proposal-regexp-v-flag

Issue: #3496, https://github.com/tc39/proposal-regexp-v-flag/issues/52

### CODE REVIEW

**Comment by mathiasbynens** [comment]
FYI @markusicu @macchiati

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  unicodeSets.call({});
+}, 'ordinary object');
+
+assert.throws(TypeError, function() {
+  unicodeSets.call([]);
+}, 'array exotic object');
+
+assert.throws(TypeError, function() {
+  unicodeSets.call(arguments);
+}, 'arguments object');
```
Comment (ptomato) [inline test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js:30]: I might suggest adding another test case for `unicodeSets.call(() => {})` in this test, its Type is Object even though its `typeof` is `"function"`.

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-non-obj.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  unicodeSets.call("string");
+}, "string");
+
+assert.throws(TypeError, function() {
+  unicodeSets.call(Symbol("s"));
+}, "symbol");
+
+assert.throws(TypeError, function() {
+  unicodeSets.call(4);
+}, "number");
```
Comment (ptomato) [inline test/built-ins/RegExp/prototype/unicodeSets/this-val-non-obj.js:40]: Similarly, for completeness consider adding a test case for bigints, e.g. `unicodeSets.call(1n)`

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-regexp.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+assert.sameValue(new RegExp(".", "u").unicodeSets, false, "new RegExp('.', 'u').unicodeSets");
+assert.sameValue(new RegExp(".", "y").unicodeSets, false, "new RegExp('.', 'y').unicodeSets");
+
+assert.sameValue(new RegExp(".", "v").unicodeSets, true, "new RegExp('.', 'v').unicodeSets");
+assert.sameValue(new RegExp(".", "vd").unicodeSets, true, "new RegExp('.', 'vd').unicodeSets");
+assert.sameValue(new RegExp(".", "vg").unicodeSets, true, "new RegExp('.', 'vg').unicodeSets");
+assert.sameValue(new RegExp(".", "vi").unicodeSets, true, "new RegExp('.', 'vi').unicodeSets");
+assert.sameValue(new RegExp(".", "vm").unicodeSets, true, "new RegExp('.', 'vm').unicodeSets");
+assert.sameValue(new RegExp(".", "vs").unicodeSets, true, "new RegExp('.', 'vs').unicodeSets");
+// Note: `new RegExp(pattern, 'vu')` throws an early parse error and is tested separately.
```
Comment (ptomato) [inline test/built-ins/RegExp/prototype/unicodeSets/this-val-regexp.js:50]: For the constructor, it's a runtime error, no? although fine to keep it in a separate file

**Review by ptomato — DISMISSED** [review]
Thanks! A couple of minor comments but this is basically OK to land.

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js (hunk 1 above)
Comment (mathiasbynens) [reply test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js:30]: Note that these tests mirror the existing tests for other RegExp flags. I’ll go ahead and address your comment here, but as a follow-up we’ll want to update those other tests as well:

image

Same for your other comment below.

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-non-obj.js (hunk 2 above)
Comment (mathiasbynens) [reply test/built-ins/RegExp/prototype/unicodeSets/this-val-non-obj.js:40]: Done

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-regexp.js (hunk 3 above)
Comment (mathiasbynens) [reply test/built-ins/RegExp/prototype/unicodeSets/this-val-regexp.js:50]: Indeed, I forgot to update this comment after copy-pasting.

**Comment by mathiasbynens** [comment]
Thanks for the review! Comments addressed.

File: test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js (hunk 1 above)
Comment (ptomato) [reply test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js:30]: Good point, I've done this in https://github.com/tc39/test262/pull/3615

**Review by ptomato — APPROVED** [review]

**Pull request merged**